File: api/src/main/java/com/dsh105/echopet/compat/api/entity/ZombieType.java
package com.dsh105.echopet.compat.api.entity;

import lombok.*;

import net.techcable.sonarpet.utils.NmsVersion;
import net.techcable.sonarpet.utils.PrettyEnum;
import net.techcable.sonarpet.utils.Versioning;

/**
 * The type of a zombie.
 *
 * NOTE: The names of these enums must match the corresponding PetData.
 */
@RequiredArgsConstructor
public enum ZombieType implements PrettyEnum {
    VILLAGER(NmsVersion.EARLIEST),
    NORMAL(NmsVersion.EARLIEST),
    HUSK(NmsVersion.v1_11_R1),
    PIGMAN(NmsVersion.EARLIEST);

    private final NmsVersion firstVersion;

    public boolean isSupported() {
        return Versioning.NMS_VERSION.compareTo(firstVersion) >= 0;
    }
}

File: api/src/main/kotlin/com/dsh105/echopet/compat/api/entity/ZombieType.kt
package com.dsh105.echopet.compat.api.entity

import lombok.*

import net.techcable.sonarpet.utils.NmsVersion
import net.techcable.sonarpet.utils.PrettyEnum
import net.techcable.sonarpet.utils.Versioning

/**
 * The type of a zombie.

 * NOTE: The names of these enums must match the corresponding PetData.
 */
@RequiredArgsConstructor
enum class ZombieType(val minimumVersion: NmsVersion = NmsVersion.EARLIEST) : PrettyEnum {
    VILLAGER,
    NORMAL,
    HUSK(minimumVersion = NmsVersion.v1_11_R1),
    PIGMAN;

    val isSupported: Boolean
        get() = Versioning.NMS_VERSION >= minimumVersion
}
